tests/test_generator.py

```diff
@@ -32,4 +34,27 @@ def test_boolean():
         "type": "boolean"
     }
     output = generate_json(schema)
+    print('result: ', output)
     assert isinstance(output, bool)
+
+
+def test_pattern_string():
+    """Test generating a pattern restricted string."""
+    # (test uses a pattern for a simple North American telephone number with an optional area code)
+    schema = {
+        "type": "string",
+        "pattern": "^(\\([0-9]{3}\\))?[0-9]{3}-[0-9]{4}$"
+    }
+    output = generate_json(schema)
+    print('result: ', output)
+    assert isinstance(output, str)
```

Comment:
Let's also check that the output validates against the regex. Use Python's standard `re` module.